### DISCUSSION TITLE
Sonoff Mini R2 - momentary switch with auto shutdown

### DISCUSSION THREAD

**Discussion by giannipi** [RavenSystem/esp-homekit-devices#1587] [MEPLHAA Scripts Collection]
**Device Name:** Sonoff Mini R2
**URL to Device:** [https://sonoff.tech/product/diy-smart-switch/minir2/](https://sonoff.tech/product/diy-smart-switch/minir2/)
**Used HAA versions:** 7.00 to 8.0.0
**Summary of functions:** Momentary switch controls on/off and automatic shutdown after 60 seconds.
**JSON:**
`{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"d":60,"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0},{"g":4}]}]}`

**Long description:**
- Light on/off on single press.
- Automatic shutdown after 60 seconds.

**Thanks to:**
[Dido Hometech](https://www.youtube.com/c/DidoHometech) flash Sonoff Mini R2 OTA
TriderG75 for the JSON

**Comment by vitoitaliano** [comment]
and if I need to set a shutdown delay of 0.5 seconds, will it work?  

JSON:
{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"**d":0.5,**"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0},{"g":4}]}]}

Or does it only work with whole seconds? Thank you.

**Comment by giannipi** [comment]
No, only whole seconds...

Il giorno gio 18 ago 2022 alle ore 13:33 vitoitaliano <
***@***.***> ha scritto:

**Reply by vitoitaliano** [reply]
So I can change by one second?  Correctly?

JSON:
{"c":{"l":13,"b":[{"g":0,"t":5}]},**"a":[{"d":1,**"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0},{"g":4}]}]}

**Comment by giannipi** [comment]
yes.

ma sei italiano?

Il giorno gio 18 ago 2022 alle ore 15:57 vitoitaliano <
***@***.***> ha scritto:

**Reply by vitoitaliano** [reply]
No. Ma abito a Verona! Ciao!🤚

**Reply by litris** [reply]
If you can use less than 1s the value you want set to 0.5s.

I hope this helps.
